src/libsystemd/sd-netlink/sd-netlink.c

```diff
@@ -494,13 +506,15 @@ static int timeout_compare(const void *a, const void *b) {
         return CMP(x->timeout, y->timeout);
 }
 
-int sd_netlink_call_async(sd_netlink *nl,
-                       sd_netlink_message *m,
-                       sd_netlink_message_handler_t callback,
-                       void *userdata,
-                       uint64_t usec,
-                       uint32_t *serial) {
-        struct reply_callback *c;
+int sd_netlink_call_async(
+                sd_netlink *nl,
+                sd_netlink_message *m,
+                sd_netlink_message_handler_t callback,
+                sd_netlink_destroy_t destroy_callback,
```

Comment:
hmm, i think sooner or later we should really rework this to have a sd_netlink_slot object similar to sd_bus_slot, and return that here instead of the serial or so, and then the destroy ptr could be set on the slot object, similar to how we do it in sd-bus...

but i figure that's a major rework, hence maybe that should be left for another time